### PR TITLE
Cast hparams to dict when not using omegaconf

### LINKED ISSUE #4542: Replace AttributeDict in with dict in checkpoint
## 🚀 Feature

When we save hyperparameters to the checkpoint, we save them in a Lightning datastructure called AttributeDict. 
We should replace this with a regular dict when saving the checkpoint.

### Motivation

Allows loading checkpoints with `torch.load` in environments where Lightning is not installed.

### Pitch

Convert AttributeDict to dict when saving, convert dict to AttributeDict when loading.

cc @Borda

### PR DESCRIPTION
**Pull request by s-rog**
## What does this PR do?

Fixes #4542

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/checkpointing/test_model_checkpoint.py [hunk 1]
```diff
@@ -27,6 +27,9 @@
 import cloudpickle
 import pytest
 import torch
+from omegaconf import Container
+from omegaconf import OmegaConf
```
Comment (Borda) [inline tests/checkpointing/test_model_checkpoint.py:31]:
```suggestion
from omegaconf import Container, OmegaConf
```

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by SkafteNicki — APPROVED** [review]
LGTM
